src/common/sentry.ts

```diff
@@ -20,73 +20,75 @@
  */
 
 import { CaptureConsole, Dedupe, Offline } from "@sentry/integrations";
-import { sentryDsn, isProduction } from "./vars";
+import * as Sentry from "@sentry/electron";
+import { sentryDsn, sentryDsnIsValid, isProduction } from "./vars";
 import { UserStore } from "./user-store";
 import logger from "../main/logger";
 
-export let sentryIsInitialized = false;
-
 /**
- * This function bypasses webpack issues.
- *
- * See: https://docs.sentry.io/platforms/javascript/guides/electron/#webpack-configuration
+ * "Translate" 'browser' to 'main' as Lens developer more familiar with the term 'main'
  */
-async function requireSentry() {
-  switch (process.type) {
-    case "browser":
-      return import("@sentry/electron/dist/main");
-    case "renderer":
-      return import("@sentry/electron/dist/renderer");
-    default:
-      throw new Error(`Unsupported process type ${process.type}`);
+function mapProcessName(processType: string) {
+  if (processType === "browser") {
+    return "main";
   }
+
+  return processType;
 }
 
 /**
  * Initialize Sentry for the current process so to send errors for debugging.
  */
-export async function SentryInit(): Promise<void> {
+export function SentryInit() {
   try {
-    const Sentry = await requireSentry();
+    const processName = mapProcessName(process.type);
 
-    try {
-      Sentry.init({
-        beforeSend: event => {
-          if (UserStore.getInstance().allowErrorReporting) {
-            return event;
-          }
+    Sentry.init({
+      beforeSend: (event) => {
+        // default to false, in case instance of UserStore is not created (yet)
+        let allowErrorReporting = false;
 
-          logger.info(`🔒  [SENTRY-BEFORE-SEND-HOOK]: allowErrorReporting: false. Sentry event is caught but not sent to server.`);
-          logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: === START OF SENTRY EVENT ===");
-          logger.info(event);
-          logger.info("🔒  [SENTRY-BEFORE-SEND-HOOK]: ===  END OF SENTRY EVENT  ===");
+        try {
+          allowErrorReporting = UserStore.getInstance()?.allowErrorReporting;
+        } catch {
+          // ignore the TypeError throw by UserStore.getInstance()
+        }
```

Comment:
You can just do `UserStore.getInstance(false)?.allowErrorReporting ?? false;`

Reply:
You mean removing the try...catch and just use `UserStore.getInstance(false)?.allowErrorReporting ?? false;` Is it possible that `getInstance` is `undefined` or not a function?

Reply:
It is a static function. Is don't think we need to be that paranoid.

Reply:
Ok, edited, thanks for confirmation https://github.com/lensapp/lens/pull/3325/commits/a0296d606deaf297bd0042170f9034009170309a